share/doc/DEVELOPERS_GUIDE.md

```diff
@@ -101,6 +102,7 @@ Code
  * Variables should have clear and explicit names.
  * Do not duplicate code.
  * Functions should have no more than 3 arguments. Otherwise create a class.
+ * Do not use raw pointers, `new` and `delete` should not be used, instead use `std::shared_ptr` or `std::unique_ptr`
```

Comment:
@JoshuaSBrown do you want to steal this comment for your next documentation update?